.github/workflows/qa.yaml

```diff
@@ -9,14 +9,48 @@ on:
       - "renovate/*"
   pull_request:
 
+# Cancel existing jobs for this branch/PR
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref_name }}
+  cancel-in-progress: true
+
 jobs:
   lint:
     uses: canonical/starflow/.github/workflows/lint-python.yaml@main
   test:
-    uses: canonical/starflow/.github/workflows/test-python.yaml@main
+    uses: canonical/starflow/.github/workflows/test-python.yaml@work/setup-vars
     with:
       fast-test-platforms: '["ubuntu-22.04", "ubuntu-24.04"]'
       fast-test-python-versions: '["3.10", "3.12", "3.13"]'
       slow-test-platforms: '["ubuntu-24.04"]'
       slow-test-python-versions: '["3.12"]'
       lowest-python-version: ""
+      pytest-markers: not java and not python and not plugin
+      setup-vars: NO_JAVA=1 NO_PYTHON=1 NO_PLUGIN=1
+  test-java-plugins:
```

Comment:
thought: I know we do not focus much on coverage, but splitting tests result in split coverage. Aside from the issue mentioned above, I wonder if we should still keep a job running everything to produce a consolidated coverage. Or we can collect the coverage files and consolidate them before uploading to codecov.